### PR TITLE
Added YAML support

### LINKED ISSUE #3248: Support ruby/rails i18n yml files
I really missed the support for ruby/rails i18n yml format. Do you have any milestone planned to support it?

Example file: https://github.com/noosfero/noosfero/blob/master/config/locales/en-US.yml

Issue first opened at https://github.com/nijel/weblate/issues/645

### PR DESCRIPTION
**Pull request by nijel**
There are two classes to support two different YAML formats:

YAMLFile : generic key/value format
RubyYAMLFile : Ruby format with target language as root node

Fixes #3248

### CODE REVIEW

File: translate/storage/test_yaml.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+import sys
+from io import BytesIO
+from translate.storage import yaml, test_monolingual
+
+
+class TestYAMLResourceUnit(test_monolingual.TestMonolingualUnit):
+    UnitClass = yaml.YAMLUnit
+
+
+class TestYAMLResourceStore(test_monolingual.TestMonolingualUnit):
```
Comment (unho) [inline translate/storage/test_yaml.py]: I guess this should be `test_monolingual.TestMonolingualStore`.

File: translate/storage/yaml.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+
+from collections import OrderedDict
+import os
+
+import six
+import yaml
+
+from translate.storage import base
+
+import yaml.constructor
```
Comment (unho) [inline translate/storage/yaml.py]: This import should be before translate toolkit one.

File: translate/storage/test_yaml.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        store.serialize(out)
+
+        assert out.getvalue() == b'''foo:
+  bar: bar
+  baz:
+    boo: booo
+'''
+
+
+class TestRubyYAMLResourceStore(test_monolingual.TestMonolingualUnit):
```
Comment (unho) [inline translate/storage/test_yaml.py]: This one should be `test_monolingual.TestMonolingualStore` too.

File: translate/storage/test_yaml.py (hunk 1 above)
Comment (unho) [inline translate/storage/test_yaml.py]: If you change this you will see that several tests fail. Please fix them.

**Comment by unho** [comment]
Very nice work, but if you correct the two misspelled classes in tests it unveils several failing tests. Please fix that.

**Comment by unho** [comment]
On a related note I was able to very quickly code some converters on top of this.

**Comment by unho** [comment]
It would be great too if you can further adjust this so comments are extracted. Here you have an example file with comments https://github.com/openstreetmap/openstreetmap-website/blob/master/config/locales/en.yml#L44-L45

File: translate/storage/test_yaml.py (hunk 1 above)
Comment (nijel) [inline translate/storage/test_yaml.py]: Oops, I've based the tests on resx and the same issue is there (translate/storage/test_resx.py).

**Comment by nijel** [comment]
Indeed supporting comments would be great, but I've decided to ignore this part for initial implementation as PyYAML support for this is poor. There is PyYAML fork which should be better (https://pypi.python.org/pypi/ruamel.yaml), but I have no experience with it.

Anyway I've addressed all issues you've pointed out (+ rebased changes on current master).

File: translate/storage/test_yaml.py (hunk 1 above)
Comment (nijel) [inline translate/storage/test_yaml.py]: See #3504 and #3505.

Comment (unho) [inline translate/storage/test_yaml.py]: Yes, thanks a lot for those PRs.

**Comment by unho** [comment]
We can live without comments for now. Landed in master.

**Comment by rvanlaak** [comment]
@unho awesome! Any news about when the next beta will be released? And @nijel how can I let Weblate install that specific beta version of `translation-toolkit`? 👍

**Comment by nijel** [comment]
You can use ``pip install 'translate-toolkit>=2.0.0b4'`` (or something similar).

**Pull request closed without merging**